conformance/lang/expressions/relational-expr/greater_than_expression.balt

```diff
@@ -0,0 +1,1283 @@
+
+const ONE = 1;
+const TWO = 2;
+type OneOrTwo ONE|TWO;
+type Ints int:Signed8|int:Unsigned32;
+
+function init() {
+   int:Signed8|int:Unsigned32? b = 13;
+   Ints? c = -23;
+   OneOrTwo? d = 1;
+
+   io:println(b > c); // @output true
+   io:println(b > d); // @output true
+   io:println(c > b); // @output false
+   io:println(c > d); // @output false
+   io:println(d > b); // @output false
+   io:println(d > c); // @output true
+}
+
+Test-Case: output
+Description: Test greater than expression when either or both operands are nil-literal
+Labels: nil-literal, optional-type, relational-expr
+
+function init() {
+   int? a = ();
+   int? b = 5;
+
+   io:println(a > b); // @output false
+   io:println(b > a); // @output false
+   io:println(() > ()); // @output false
+}
+
+Test-Case: output
+Description: Test greater than expression when either or both operands are boolean
+Labels: boolean, relational-expr
+
+function init() {
+   boolean a = false;
+   boolean b = true;
+   io:println(a > b); // @output false
+   io:println(b > a); // @output true
+   io:println(a > a); // @output false
+   io:println(b > b); // @output false
+}
+
+Test-Case: output
+Description: Test greater than expression when both operands are optional boolean
+Labels: boolean, optional-type, relational-expr
+
+function init() {
+   boolean? a = false;
+   boolean? b = true;
+   io:println(a > b); // @output false
+   io:println(b > a); // @output true
+   io:println(a > a); // @output false
+   io:println(b > b); // @output false
+}
+
+Test-Case: output
+Description: Test greater than expression when either or both operands are string
+Labels:relational-expr, string
+
+function init() {
+   string a = "ABCD";
+   string b = "ABCDE";
+   string c = "DCBA";
+   io:println(a > b); // @output false
+   io:println(b > a); // @output true
+   io:println(a > c); // @output false
+   io:println(c > a); // @output true
+   io:println(c > c); // @output false
+}
+
+Test-Case: output
+Description: Test greater than expression when one operand is optional string
+Fail-Issue: ballerina-platform/ballerina-lang#35195
+Labels:relational-expr, optional-type, string
+
+function init() {
+   string? a = "ABCD";
+   string b = "ABCDE";
+   string? c = "DCBA";
+   io:println(a > b); // @output false
+   io:println(b > a); // @output true
+}
+
+Test-Case: output
+Description: Test greater than expression when both operands are optional string
+Labels: relational-expr, optional-type, string
+
+function init() {
+   string? a = "ABCD";
+   string? b = "ABCDE";
+   string? c = "DCBA";
+   io:println(a > b); // @output false
+   io:println(b > a); // @output true
+   io:println(a > c); // @output false
+   io:println(c > a); // @output true
+   io:println(c > c); // @output false
+}
+
+Test-Case: output
+Description: Test greater than expression when both operands are positive decimals
+Labels: decimal, DecimalFloatingPointNumber,  DecimalNumber, relational-expr
```

Comment:
```suggestion
Labels: decimal, DecimalFloatingPointNumber, DecimalNumber, relational-expr
```
There's additional space in some of the places where `DecimalNumber` label has been used. Let's fix them too.